# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/ScriptableContentIterator.h
# Commit: 3601308f553b
# Full Hash: 3601308f553b3d381a68f6e90a971bd7cb7b1684
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2023-08-10 04:42:07
# Regressor Bug: 1845215
# File Overlap Count: 1
# Description:
#   Bug 1845215 - part 1: Make `ContentIteratorBase` a template class to work without strong pointers r=smaug
#   
#   `ContentEventHandler` works with `PostContentIterator` and `PreContentIterator`
#   when it scans DOM nodes in a range.  While iterating the DOM nodes, script
#   never runs.  Therefore, we can make `ContentEventHandler` work with new
# ==============================================================================

diff -r f2506aaa10bd -r 3601308f553b dom/base/ScriptableContentIterator.h
--- a/dom/base/ScriptableContentIterator.h	Wed Aug 09 23:05:26 2023 +0000
+++ b/dom/base/ScriptableContentIterator.h	Thu Aug 10 00:59:28 2023 +0000
@@ -27,7 +27,7 @@
   void EnsureContentIterator();
 
   IteratorType mIteratorType;
-  UniquePtr<ContentIteratorBase> mContentIterator;
+  UniquePtr<SafeContentIteratorBase> mContentIterator;
 };
 
 }  // namespace mozilla